docs/spec/data-plane.md

```diff
@@ -0,0 +1,71 @@
+# Knative Eventing Data Plane Contracts
+
+## Introduction
+
+Developers using Knative eventing need to know what is supported for delivery to user provided components that receive events. Knative Eventing defines contract for data plane components and we have listed them here.
+
+## Data plane contract for Sinks
+
+A **Sink** is an [_addressable_](./interfaces.md#addressable) resource that takes responsibility for the event. A Sink could be a consumer of events, or middleware. A Sink MUST be able to receive CloudEvents over HTTP and HTTPS.
+
+A **Sink** MAY be [_callable_](./interfaces.md#callable) resource that represents an Addressable endpoint which receives an event as input and optionally returns an event to forward downstream
+
+Almost every component in Knative Eventing may be a Sink providing composability.
+
+Every Sink MUST support HTTP Protocol Binding for CloudEvents [version 1.0](https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md) and [version 0.3](https://github.com/cloudevents/spec/blob/v0.3/http-transport-binding.md).
+
+### Details of HTTP Support
+
+This section adds restrictions on [requirements in HTTP Protocol Binding for CloudEvents](https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md#12-relation-to-http).
+
+Sinks MUST reject all HTTP requests with a method other than
+POST responding with HTTP status code `405 Method Not Supported`. Non-event requests (e.g. health checks) are not constrained.
```

Comment:
PUT?  (see CE spec)

Reply:
I though that only HTTP POST was used in Knative Eventing sources and channels?
The HTTP binding does not specifiy [what HTTP methods are supported leaving it open](https://github.com/cloudevents/spec/blob/master/http-protocol-binding.md#12-relation-to-http) so what Sink developers should expect?

Channels I think currently only support POST so that would be good common denominator? https://github.com/knative/eventing-contrib/blob/60db97173512ac73985946081db2234d563413f5/pkg/channel/message_receiver.go#L85
https://github.com/knative/eventing-contrib/blob/60db97173512ac73985946081db2234d563413f5/pkg/channel/message_receiver.go#L126
https://github.com/knative/eventing-contrib/blob/17bbb2505a231cb017b6e8422b8eada7c791b93d/vendor/knative.dev/eventing/pkg/channel/event_receiver.go#L132

Reply:
I think it may be best resolved by opening issue for it? https://github.com/knative/eventing/issues/2657